src/pages/iou/IOUModal.js

```diff
@@ -74,11 +75,21 @@ class IOUModal extends Component {
     }
 
     componentDidUpdate(prevProps) {
-        // Dismiss modal when previous iousReport changes from the new iousReport
-        // This should occur the data changes from within the modal activity
-        if (!_.isEqual(prevProps.iousReport, this.props.iousReport)) {
+        // Dismiss modal when the length of transactions for any iousReport changes
+        if (!prevProps.iousReport) { return; }
+
+        if (this.state.isTransactionComplete === true) {
             return Navigation.dismissModal();
         }
+
+        // Object.keys(this.props.iousReport)
```

Comment:
@Julesssss  This didn't work with Onyx.mergeCollection() so I instead opted to use state.isTransactionComplete to figure out when to close the modal. What do you suggest I should do? 

The logic works perfectly well in the case of Request, when I merge just a single report key, but when I merge the entire object, it doesn't get called. I'm thinking it's mutability issue, and I need to look more into it. But just wanted to know if it was okay to instead use state to know when to close the modal?

Reply:
Oh, so you didn't get notified about the report when mergeCollection was called?  Odd.

We want to avoid returning from an Actions (Promises or callbacks) so I'd suggest looking into the reason that mergeCollection doesn't work first before we look for alternatives. I'll take a look at this issue tomorrow and let you know if I find anything.

Reply:
I didn't find any time for this today unfortunately. Will try again tomorrow. Any luck on your side?

Reply:
@Julesssss I faced the same issue when I looked at it briefly today morning. Will take a deeper look now!

Reply:
Didn't really figure out anything :/

Reply:
@barun1997 - Would you mind providing an example of the `Onyx.mergeCollection()` code you were having issues with? We have a [test in Onyx to merge objects](https://github.com/Expensify/react-native-onyx/blob/master/tests/unit/onyxTest.js#L163-L241), so I am wondering if we can confirm a bug easily using that unit test. Possibly by using a more complicated object like you are using? Happy to help here!

Reply:
Hi @AndrewGable Thank you!  I have pushed a new commit which includes the updated code.

Reply:
This was a bug in Onyx, which [has just been fixed](https://github.com/Expensify/react-native-onyx/pull/57). We've just [bumped](https://github.com/Expensify/Expensify.cash/pull/2200) the Onyx version in Expensify.cash, so you should be able to resolve your issue after merging master.

Reply:
@Julesssss Got it, thank you!

Reply:
Hi @Julesssss, just to give you an update, the issue still seems to persist on my end. However, I will look into it more to see if the bug is with how I've used mergeCollection. Please feel free to give suggestions if you have any.

It seems to work with IOURequest, like before where just merge is used. I used _.isEqual for now -- just to make sure the componentDidUpdate implementation wasn't the problem.

I've attached a video for the two instances below: 

https://user-images.githubusercontent.com/19339044/113383882-93e35180-93a4-11eb-9b13-ddeee03ae957.mov

Reply:
Hi @barun1997, we're now tracking [this here](https://github.com/Expensify/react-native-onyx/pull/60).

Reply:
@Julesssss Thank you Jules, I will take a look.